MigrationScript/Migration.ps1

```diff
@@ -55,81 +70,91 @@ function Main
+			# if the hana provider is using key vault to fetch user credentials, skip the migration. 
+			# (To be handled later, once the feature is enabled in ams v2)
+			if(!$secret.properties.hanaDbPasswordKeyVaultUrl)
+			{                  
+				$logger.LogInfoObject("Trying to migrate SapHana Provider", $secret.name);
+				$hanaMigrationResult = MigrateHanaProvider -secretName $secret.name -secretValue $secret -logger $logger
+
+				$requestHana = @{
+					name = $secret.name
+					type = $secret.type
+					state = $hanaMigrationResult.provisiongState
+				}
 
-					if($hanaMigrationResult.provisiongState -eq "Succeeded"){
-						$logger.LogInfoObject("Adding the following transformed SapHana object to migration list", $requestHana)
-				   	}
+				if($hanaMigrationResult.provisiongState -eq "Succeeded"){
+					$logger.LogInfoObject("Adding the following transformed SapHana object to migration list", $requestHana)
+				}
 
-                    $saphanaTransformedList.Add($requestHana) | Out-Null
-               }
-               else
-               {
-                    $request = @{
-                        name = $secret.name
-                        type = $secret.type
-						state = "Unsupported"
-                    }
-					
-                    $unsupportedProviderList.Add($request) | Out-Null
-                    $logger.LogError("Unsupported Type - SapHana Integrated KeyVault",
-						"100",
-						"Please wait for the support to be enabled in AMSv2 to migrate provider - " + $secret.name)                  
-               }
+				$saphanaTransformedList.Add($requestHana) | Out-Null
+			}
+			else
+			{
+				$request = @{
+					name = $secret.name
+					type = $secret.type
+					state = "Unsupported"
+				}
+				
+				$unsupportedProviderList.Add($request) | Out-Null
+				$logger.LogError("Unsupported Type - SapHana Integrated KeyVault",
+					"100",
+					"Please wait for the support to be enabled in AMSv2 to migrate provider - " + $secret.name)                  
+			}
         }
         elseif($secret.type -like "sapnetweaver" -and ($providerType -like "sapnetweaver" -or $providerType -like "all"))
         {
             # if the netweaver provider is using key vault to fetch user credentials, skip the migration. 
+			# and if the provider is a RFC SAP Netweaver provider then skip the migration. 
 			# (To be handled later, once the feature is enabled in ams v2)
-            if(!$secret.properties.sapPasswordKeyVaultUrl)
+            if(!$secret.properties.sapPasswordKeyVaultUrl -and !$secret.properties.sapRfcSdkBlobUrl)
             {
-				$hashTable = ParseSapNetWeaverHostfile -fileName "hosts.json"
+				$hashTable = ParseSapNetWeaverHostfile -fileName "hosts.json" -logger $logger;
 
                 $logger.LogInfoObject("Trying to migrate SapNetWeaver Provider", $secret.name);
 				$netweaverMigrationResult = MigrateNetWeaverProvider -secretName $secret.name -secretValue $secret -hostfile $hashTable[$secret.name] -logger $logger
```

Comment:
If secret name doesn't exist in hostfile then we were getting error in put provider scenario.We can add a check here to avoid breaking PUT provider of AMSv2

Reply:
Added the code for same. Resolving the issue